Update build instructions (#141)
qutip/qutip-doc: README.md

For pre-built documentation, see http://www.qutip.org/documentation.html

Building
--------

The main Python requirements for the documentation are `sphinx`, `sphinx-gallery`, `sphinx_rtd_theme`, `numpydoc` and `ipython`.
You should build or install the version of QuTiP you want to build the documentation against int the same environment.
You will also need a sensible copy of `make`, and if you want to build the LaTeX documentation then also a `pdflatex` distribution.
As of 2021-04-20, the `conda` recipe for `sphinx_rtd_theme` is rather old compared to the `pip` version, so it's recommended to use a mostly `pip`-managed environment to do the documentation build.

The simplest way to get a functional build environment is to use the `requirements.txt` file in this repository, which completely defines a known-good `pip` environment (tested on Python 3.8, but not necessarily limited to it).
If you typically use conda, the way to do this is
```bash
$ conda create -n qutip-doc-build python=3.8
$ conda activate qutip-doc-build
$ pip install -r /path/to/qutip-doc/requirements.txt
```

> You will also need to build or install the main QuTiP library in the same environment.

Reviewer: This needs to be moved to a line before `qutip-doc` requirements are installed otherwise there's errors related to some packages not being installed and dependency issues.  

```suggestion
You will first need to [build the environment for QuTip](http://qutip.org/docs/4.1/installation.html#building-your-conda-environment) and then [install main QuTiP library](http://qutip.org/docs/4.1/installation.html#installing-from-source) in the same environment.
```

Author: You shouldn't need to build the environment for QuTiP to run the lines I wrote.  You'll need to _afterwards_ to build the docs, but the exact steps you take depend on what you want to build docs for - technically you can build the docs in a working build with QuTiP installed by `pip install qutip`, and this requirements file will have satisfied everything needed to do that.  Also, your suggested documentation links are very very out of date - you might want to check that you're using the latest versions personally.

If you get errors running the commands I posted, can you post the errors?

Reviewer: > Also, your suggested documentation links are very very out of date

Thanks for pointing this out ! I have switched to the latest version. I think with this version's installation instructions, I do not get any errors. 

I felt that you would need to build the environment in order to use `pip install qutip` as well ?

Author: Nope, as of 4.6 we distribute pre-compiled wheels so `pip install qutip` doesn't need to build anything locally any more (similar to how `conda install` has always worked for us).

The thing is that if you want to modify the main repository as well, you need to follow the steps to build from source instead, though right now, the `requirements.txt` file in this repo actually installs everything you need to do that anyway (a minor coincidence).

Reviewer: As an aside, do you mind if I create a PR for enabling versions in the documentation sidebar ? Say if I am [on a page](http://qutip.org/docs/4.1/guide/guide-bloch.html) where there's no instructions on how to [switch to the latest version](http://qutip.org/docs/4.1/frontmatter.html), a sidebar could help showing version numbers and the ability to switch between these. 

I found a [Sphinx based extension](https://github.com/sphinx-contrib/sphinxcontrib-versioning) that could allow this (although I haven't looked at in detail yet).

Author: We're certainly interested in this sort of thing, but there might be quite a lot of fiddly testing required.  The documentation section of qutip.org doesn't point to this repository, but just to a subfolder of the qutip.github.io repository, which has no knowledge of the git history of this repo.  At the moment, working with the website can be quite a faff because the `.git` directory is very large in part because we haven't handled the documentation history very well.

Reviewer: So, instead of using the extension I found, I could create a tab or link somewhere in the sidebar, header or footer linking to [this page on the website](http://qutip.org/documentation.html) ?

Author: Honestly, I have no real idea what's possible with the extension you linked - if that's a sensible way to do it, we're happy in principle to accept a PR doing it.

The best way to go about this might be to create an issue in this repository that describes how all of this would work (might need a little bit of prototype code to test), including how it will work when deployed to the website.  You might need to fork the qutip.github.io repository and build your own GitHub pages site from it to test these things out.

Reviewer: Thanks ! I myself don't understand how the extension works either (I was planning on taking a look at it later). But based on what you said about git history, I thought an easier solution could be to add a link somewhere in the template for the webpage linked above along the lines of `here's how you can access links to other versions of QuTip`. 

I will try to upload a PR in the proper repo soon. We could continue the discussion there i.e. where the link should appear on the documentation page.
